capella_console_client/tasking.py

```diff
@@ -1,13 +1,133 @@
+    geometry: geojson.geometry.Geometry,
+    name: Optional[str] = "",
+    description: Optional[str] = "",
+    window_open: Optional[Union[datetime, str]] = None,
+    window_close: Optional[Union[datetime, str]] = None,
+    collection_tier: Optional[Union[str, CollectionTier]] = CollectionTier.low,
+    product_category: Optional[Union[str, ProductClass]] = ProductClass.standard,
+    archive_holdback: Optional[Union[str, ArchiveHoldback]] = ArchiveHoldback.none,
+    custom_attribute_1: Optional[str] = None,
+    custom_attribute_2: Optional[str] = None,
+    product_types: Optional[List[Union[str, ProductType]]] = None,
+    # Collect constraints
+    collect_mode: Optional[Union[str, InstrumentMode]] = InstrumentMode.spotlight,
+    look_direction: Optional[Union[str, ObservationDirection]] = ObservationDirection.either,
+    asc_dsc: Optional[Union[str, OrbitState]] = OrbitState.either,
+    orbital_planes: Optional[List[Union[int, OrbitalPlane]]] = None,
+    local_time: Optional[List[int]] = None,
+    off_nadir_min: Optional[int] = None,
+    off_nadir_max: Optional[int] = None,
+    elevation_min: Optional[int] = None,
+    elevation_max: Optional[int] = None,
+    image_length: Optional[int] = None,
+    image_width: Optional[int] = None,
+    azimuth: Optional[int] = None,
+    grr_min: Optional[int] = None,
+    grr_max: Optional[int] = None,
+    srr_min: Optional[int] = None,
+    srr_max: Optional[int] = None,
+    azr_min: Optional[int] = None,
+    azr_max: Optional[int] = None,
+    nesz_max: Optional[int] = None,
+    num_looks: Optional[int] = None,
+    polarization: Optional[Union[str, Polarization]] = None,
+) -> Dict[str, Any]:
+
+    if isinstance(window_open, str):
+        try:
+            window_open = parse(window_open)
+        except ParserError:
+            raise ValueError("Could not parse window_open string into a useable datetime")
+    if isinstance(window_close, str):
+        try:
+            window_close = parse(window_close)
+        except ParserError:
+            raise ValueError("Could not parse window_close string into a useable datetime")
+
+    if not window_open:
+        window_open = datetime.utcnow()
+    if not window_close:
+        window_close = window_open + timedelta(days=7)
+
+    window_open = window_open.isoformat(timespec="milliseconds") + "Z"
+    window_close = window_close.isoformat(timespec="milliseconds") + "Z"
+
+    loc = locals()
+    collect_constraints = {_snake_to_camel(k): loc[k] for k in COLLECT_CONSTRAINTS_KEYS if k in loc and loc[k]}
+
+    payload = {
+        "type": "Feature",
+        "geometry": geometry,
+        "properties": {
+            "taskingrequestName": name,
+            "taskingrequestDescription": description,
+            "windowOpen": window_open,
+            "windowClose": window_close,
+            "collectionTier": collection_tier,
+            "productCategory": product_category,
+            "archiveHoldback": archive_holdback,
```

Comment:
I'd make `archiveHoldback`, `customAttribute1` and `customAttribute2` conditional

something like

```python
# above list
CONDITIONAL_PROPERTIES = [..]

# might want to default archive holdback to None
cond_properties = {_snake_to_camel(k): loc[k] for k in CONDITIONAL_PROPERTIES if k in loc and loc[k] is None}

if cond_properties:
    payload["properties"].update(cond_properties)
```